addons/stock/static/src/stock_forecasted/stock_forecasted.js

```diff
@@ -51,9 +51,14 @@ export class StockForecasted extends Component {
         //Following is used as a fallback when the forecast is not called by an action but through browser's history
         if (!this.resModel) {
             if (this.props.action.res_model) {
-                const actionModel = await this.orm.read('ir.model', [Number(this.props.action.res_model)], ['model']);
-                if (actionModel[0]?.model) {
-                    this.resModel = actionModel[0].model
+                const res_model_id = Number(this.props.action.res_model);
+                if (!isNaN(res_model_id)) {
+                    const actionModel = await this.orm.read('ir.model', [res_model_id], ['model']);
+                    if (actionModel[0]?.model) {
+                        this.resModel = actionModel[0].model;
+                    }
+                } else {
+                    this.resModel = this.props.action.res_model;
```

Comment:
I don't see in which case `this.props.action.res_model` would be a number ?
Ins't this enough ?
```suggestion
                this.resModel = this.props.action.res_model;
```
Or do you achieve to get an ID in another way ?